### PR TITLE
Failure injection example script

### PR DESCRIPTION
**Pull request by potaito**
Very basic python example for failure injection. Drone takes off, flies to a higher altitude and then receives a GPS failure injection, upon which it will failsafe and land.

Tested with px4_sitl / jmavsim

### CODE REVIEW

File: examples/failure_injection.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    flying_alt = goto_alt + 20.0 # To fly drone 20m above the ground plane
+    await drone.action.goto_location(goto_lat, goto_lon, flying_alt, 0)
+
+    await asyncio.sleep(5)
+
+    print("-- Injecting GPS failure")
+    await drone.failure.inject(FailureUnit.SENSOR_GPS, FailureType.OFF, instance=0)
+
+    print("-- Waiting 20s before exiting script...")
+    await asyncio.sleep(20)
```
Comment (julianoes) [inline examples/failure_injection.py:56]: You could consider resetting the param at the end here.

**Review by julianoes — COMMENTED** [review]
Amazing, thanks for that!

File: examples/failure_injection.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    goto_lat = 0.0
+    goto_lon = 0.0
+    goto_alt = 0.0
+    async for position in drone.telemetry.position():
+        # Only need position once
+        if position.latitude_deg and position.longitude_deg:
+            goto_lat = position.latitude_deg
+            goto_lon = position.longitude_deg
+            goto_alt = position.absolute_altitude_m
+            break
```
[the remark this answers is not in the document]
Comment (potaito) [reply examples/failure_injection.py:44]: I was wondering that as well :sweat_smile:

File: examples/failure_injection.py (hunk 1 above)
Comment (potaito) [reply examples/failure_injection.py:56]: Right, will do :+1:

File: examples/failure_injection.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JonasVautherin) [reply examples/failure_injection.py:44]: I don't see how, without putting it in a function, like:

```python
goto_lat, goto_lon, goto_alt = await fetch_position()
```

And the function looking like:

```python
async def fetch_position():
    async for position in drone.telemetry.position():
        if position.latitude_deg and position.longitude_deg:
            return position.latitude_deg, position.longitude_deg, position.absolute_altitude_m
```

[the remark this answers is not in the document]
Comment (julianoes) [reply examples/failure_injection.py:44]: Ok, so we'll leave it.

**Review by julianoes — APPROVED** [review]

**Pull request merged**